GetStream/stream-chat-android
# [3111] Fix channel unread count for muted channels

### 🎯 Goal

Currently we increment individual channel unread count even if the user has muted the channel in question. This is not in line with the [docs](https://getstream.io/chat/docs/android/muting_channels/?language=kotlin&q=mute). 

This PR takes channel mutes into account and does not increment the unread count for individual channels if the channel in question has been muted.

### 🛠 Implementation details

- Added an extra check to see if the channel is muted
- Switched uses of `chatDomainImpl` to `GlobalMutatedState.getOrCreate()` 

### 🧪 Testing

- Use two instances of the app on separate phones/ emulators.
- Send messages from User A to User B while the the channel is muted. **(Unread count for said channel should go up)**
- Mute the channel in question in User B's app and send messages from User A's app **(Unread count for said channel should stay the same as before)**

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

![counting](https://user-images.githubusercontent.com/37080097/155291309-b9972133-9abe-4ecd-a91e-b03602671394.gif)
